src/Parser.php

```diff
@@ -1025,6 +997,71 @@ private function parseStringLiteralExpression2($parentNode) {
         return $expression;
     }
 
+    /**
+     * Double-quoted and heredoc strings support a basic set of expression types, described in http://php.net/manual/en/language.types.string.php#language.types.string.parsing
+     * Supported: $x, $x->p, $x[0], $x[$y]
+     * Not supported: $x->p1->p2, $x[0][1], etc.
+     * Since there is a relatively small finite set of allowed forms, I implement it here rather than trying to reuse the general expression parsing code.
+     */
+    private function parseTemplateStringExpression($parentNode) {
+        $token = $this->getCurrentToken();
+        if ($token->kind === TokenKind::VariableName) {
+            $var = $this->parseSimpleVariable($parentNode);
+            $token = $this->getCurrentToken();
+            if ($token->kind === TokenKind::OpenBracketToken) {
+                return $this->parseTemplateStringSubscriptExpression($var);
+            } else if ($token->kind === TokenKind::ArrowToken) {
+                return $this->parseTemplateStringMemberAccessExpression($var);
+            } else {
+                return $var;
+            }
+        }
+
+        return null;
+    }
+
+    private function parseTemplateStringSubscriptExpression($postfixExpression) : SubscriptExpression {
+        $subscriptExpression = new SubscriptExpression();
+        $subscriptExpression->parent = $postfixExpression->parent;
+        $postfixExpression->parent = $subscriptExpression;
+
+        $subscriptExpression->postfixExpression = $postfixExpression;
+        $subscriptExpression->openBracketOrBrace = $this->eat(TokenKind::OpenBracketToken); // Only [] syntax is supported, not {}
```

Comment:
What do you mean with the comment?

`"echo {$myFancyArray[5]}"` is valid too, and will output the sixth element of the array called `myFancyArray`,